node/node.go

```diff
@@ -801,7 +799,33 @@ func (n *Node) superviseManager(ctx context.Context, securityConfig *ca.Security
 		timer := time.NewTimer(16 * time.Second)
 		select {
 		case <-timer.C:
-			log.G(ctx).Warn("failed to get worker role after manager stop, restarting manager")
+			if wasRemoved {
+				// We need to be extra careful about restarting
```

Comment:
Could we update this comment with why this needs to be re-forced?  I was a little confused about what an extra write to `forceCertRenewal` would do, but your explanation that it was never written to because `n.notifyNodeChange` didn't receive a message in time makes this make a lot more sense.